includes/admin/tools/data/class-give-tools-delete-import-donors.php

```diff
@@ -0,0 +1,442 @@
+	 */
+	public function get_data() {
+
+		// Get the donation id's.
+		$donation_ids = $this->get_option( $this->donation_key );
+
+		/**
+		 * Return false id not Import donation is found.
+		 */
+		if ( empty( $donation_ids ) ) {
+			$this->is_empty   = true;
+			$this->total_step = 1;
+
+			return false;
+		}
+
+		// Get the current step.
+		$step = (int) $this->get_step();
+
+		// get teh donor ids.
+		$donor_ids = $this->get_option( $this->donor_key );
+
+		// In step to we delete all the donation in loop.
+		if ( 2 === $step ) {
+			$pass_to_donor = false;
+			$page          = (int) $this->get_step_page();
+			$page ++;
+			$count = count( $donation_ids );
+
+			$this->total_step     = ( ( count( $donation_ids ) / $this->per_step ) * 2 ) + count( $donor_ids );
+			$this->step_completed = $page;
+
+
+			if ( $count > $this->per_step ) {
+
+				$this->update_option( $this->step_on_key, $page );
+				$donation_ids = $this->get_delete_ids( $donation_ids, $page );
+				$current_page = (int) ceil( $count / $this->per_step );
+
+				if ( $page === $current_page ) {
+					$pass_to_donor = true;
+				}
+			} else {
+				$pass_to_donor = true;
+			}
+
+			if ( true === $pass_to_donor ) {
+				$this->update_option( $this->step_key, 'donor' );
+				$this->update_option( $this->step_on_key, '0' );
+			}
+
+			foreach ( $donation_ids as $item ) {
+				wp_delete_post( $item, true );
+			}
+		}
+
+
+		// Here we delete all the donor
+		if ( 3 === $step ) {
+			$page  = (int) $this->get_step_page();
+			$count = count( $donor_ids );
+
+			$this->total_step     = ( ( count( $donation_ids ) / $this->per_step ) * 2 ) + count( $donor_ids );
+			$this->step_completed = $page + ( count( $donation_ids ) / $this->per_step );
+
+			$args = apply_filters( 'give_tools_reset_stats_total_args', array(
+				'post_type'      => 'give_payment',
+				'post_status'    => 'any',
+				'posts_per_page' => 1,
+				'meta_key'       => '_give_payment_mode',
+				'meta_value'     => 'live',
+				'author'         => $donor_ids[ $page ]
+			) );
+
+			$donation_posts = get_posts( $args );
+			if ( empty( $donation_posts ) ) {
+				Give()->donors->delete_by_user_id( $donor_ids[ $page ] );
+			}
+
+			$page ++;
+			$this->update_option( $this->step_on_key, $page );
+			if ( $count === $page ) {
+				$this->is_empty = false;
+
+				return false;
+			}
+
+			return true;
+		}
+
+		return true;
+	}
+
+	public function get_delete_ids( $donation_ids, $page ) {
+		$index            = $page --;
+		$count            = count( $donation_ids );
+		$temp             = 0;
+		$current_page     = 0;
+		$post_delete      = $this->per_step;
+		$page_donation_id = array();
+
+		foreach ( $donation_ids as $item ) {
+			$temp ++;
+			$page_donation_id[ $current_page ][] = $item;
+			if ( $temp === $post_delete ) {
+				$current_page ++;
+				$temp = 0;
+			}
+		}
+
+		return $page_donation_id[ $page ];
+	}
+
+	/**
+	 * Given a key, get the information from the Database Directly
+	 *
+	 * @since  1.8.12
```

Comment:
@raftaar1191 is this wrong `@since` tag, please update them if any

Reply:
Done